scripts/globals/events/starlight_celebrations.lua

```diff
@@ -0,0 +1,943 @@
+    local contentEnabled = xi.events.starlightCelebration.isStarlightEnabled()
+    local item = trade:getItemId()
+    local npcID = tostring(npc:getID())
+    local npcTrades = player:getLocalVar("[Smilebringers][" .. npcID .. "]GiftsReceived")
+    local head = player:getEquipID(xi.slot.HEAD)
+    ------------------
+    -- 2007 Edition --
+    ------------------
+    if (contentEnabled == 1) then
+        ----------------------
+        -- Presents Allowed --
+        ----------------------
+        local gifts_table =
+        {
+            1742, -- Present for Kiddies
+            1743, -- Present for Kiddies
+            1744, -- Present for Kiddies
+            1745, -- Present for Kiddies
+            4215, -- Popstar
+            4216, -- Brilliant Snow
+            4217, -- Sparkling Hand
+            4218, -- Air Rider
+            4167, -- Cracker
+            4168, -- Twinkle Shower
+            4169, -- Little Comet
+        }
+
+        local presents_table =
+        {
+            1742,
+            1743,
+            1744,
+            1745,
+        }
+        ---------------
+        -- Hat Check --
+        ---------------
+        if (head == 15179 or head == 15178) then
+            if (npcTrades < 14) then
+                for itemInList = 1, #gifts_table do
+                    if (item == gifts_table[itemInList] and (head == 15179 or head == 15178) and (item == presents_table[itemInList]) and (player:getFameLevel(xi.quest.fame_area.HOLIDAY) < 9)) then
+                        player:showText(npc, ID.text.GIFT_THANK_YOU)
+                        player:messageSpecial(ID.text.BARRELS_JOY_TO_CHILDREN)
+                        player:addFame(xi.quest.fame_area.HOLIDAY, 128)
+                        player:tradeComplete()
+                    elseif (item == gifts_table[itemInList] and (head == 15179 or head == 15178) and (player:getFameLevel(xi.quest.fame_area.HOLIDAY) < 9)) then
+                        player:showText(npc, ID.text.GIFT_THANK_YOU)
+                        player:messageSpecial(ID.text.JOY_TO_CHILDREN)
+                        player:addFame(xi.quest.fame_area.HOLIDAY, 32)
+                        player:tradeComplete()
+                    elseif ((item == gifts_table[itemInList]) and (head == 15179 or head == 15178) and (player:getFameLevel(xi.quest.fame_area.HOLIDAY) >= 9)) then
+                        player:showText(npc, ID.text.ONLY_TWO_HANDS)
+                    end
+                end
+                player:setLocalVar("[Smilebringers][" .. npcID .. "]GiftsReceived", npcTrades + 1)
+            else
+                player:showText(npc, ID.text.ONLY_TWO_HANDS)
+            end
+        else
+            return
+        end
+    end
+end
+
+--------------------------------
+-- Gifts for NPCs Sub-Quest --
+--------------------------------
+
+function xi.events.starlightCelebration.npcGiftsMoogleOnTrigger(player)
+    if xi.events.starlightCelebration.isStarlightEnabled ~= 0 then
```

Comment:
Does this need a default event? If no, can this be an early return?

Reply:
Most likely, will adjust!

Reply:
Updated!